src/main/obsRecorder.ts

```diff
@@ -300,48 +304,53 @@ const watchVideoSourceSize = (sceneItem: ISceneItem, videoSource: IInput, baseRe
 /*
 * setupSources
 */
-const setupSources = (scene: any, audioInputDeviceId: string, audioOutputDeviceId: string ) => {
+const setupSources = () => {
+  const { audioInputDeviceId, audioOutputDeviceId } = recorderOptions;
+
   clearSources();
 
-  osn.Global.setOutputSource(1, scene);
+  osn.Global.setOutputSource(currentObsTrack++, scene);
 
   setSetting('Output', 'Track1Name', 'Mixed: all sources');
-  let currentTrack = 2;
 
   getAvailableAudioInputDevices()
-    .forEach(device => {
-      const source = osn.InputFactory.create('wasapi_input_capture', 'mic-audio', { device_id: device.id });
-      setSetting('Output', `Track${currentTrack}Name`, device.name);
-      source.audioMixers = 1 | (1 << currentTrack-1); // Bit mask to output to only tracks 1 and current track
-      source.muted = audioInputDeviceId === 'none' || (audioInputDeviceId !== 'all' && device.id !== audioInputDeviceId);
-      console.log(`[OBS] Selecting audio input device: ${device.name} ${source.muted ? ' [MUTED]' : ''}`)
-      osn.Global.setOutputSource(currentTrack, source);
-      source.release()
-      currentTrack++;
-    });
+    .filter(device => audioInputDeviceId !== 'none' && (audioInputDeviceId === 'all' || device.id === audioInputDeviceId))
+    .forEach(device => addAudioSource(currentObsTrack++, 'wasapi_input_capture', device));
 
   getAvailableAudioOutputDevices()
-    .forEach(device => {
-      const source = osn.InputFactory.create('wasapi_output_capture', 'desktop-audio', { device_id: device.id });
-      setSetting('Output', `Track${currentTrack}Name`, device.name);
-      source.audioMixers = 1 | (1 << currentTrack-1); // Bit mask to output to only tracks 1 and current track
-      source.muted = audioOutputDeviceId === 'none' || (audioOutputDeviceId !== 'all' && device.id !== audioOutputDeviceId);
-      console.log(`[OBS] Selecting audio output device: ${device.name} ${source.muted ? ' [MUTED]' : ''}`)
-      osn.Global.setOutputSource(currentTrack, source);
-      source.release()
-      currentTrack++;
-    });
+    .filter(device => audioOutputDeviceId !== 'none' && (audioOutputDeviceId === 'all' || device.id === audioOutputDeviceId))
+    .forEach(device => addAudioSource(currentObsTrack++, 'wasapi_output_capture', device));
 
-  setSetting('Output', 'RecTracks', parseInt('1'.repeat(currentTrack-1), 2)); // Bit mask of used tracks: 1111 to use first four (from available six)
-}
+  setRecTracks();
+};
 
 /**
- * Clear all sources from the global output of OBS
+ * Set the Output.RecTracks setting for OBS
  */
- const clearSources = (): void => {
-  console.log("[OBS] Removing all output sources")
+const setRecTracks = (): void => {
+  if (currentObsTrack == 1) {
+    setSetting('Output', 'RecTracks', 0);
+    return;
+  }
 
-  // OBS allows a maximum of 64 output sources
+  // Bit mask of used tracks: 1111 to use first four (from available six)
+  setSetting('Output', 'RecTracks', parseInt('1'.repeat(currentObsTrack - 1), 2));
```

Comment:
this is fairly tricky to follow. Can we split this into two lines?

```
const audioBitMask = parseInt('1'.repeat(currentObsTrack - 1), 2);
setSetting('Output', 'RecTracks', audioBitMask);
```

Reply:
This is literally just a copied line, it was already there to behin with :p

Reply:
I did split it up, though, since you asked for it.

Reply:
Ah, hadn't realised. Apologies!